core/src/main/java/org/apache/spark/util/collection/unsafe/sort/UnsafeSorterSpillWriter.java

```diff
@@ -62,6 +62,8 @@ public UnsafeSorterSpillWriter(
       int fileBufferSize,
       ShuffleWriteMetrics writeMetrics,
       int numRecordsToWrite) throws IOException {
+    // Space used by prefix + len + recordLength is more than 4 + 8 bytes
+    assert (diskWriteBufferSize > 12);
```

Comment:
assert shouldn't be used to check arguments in public APIs. But, despite its visibility I'm not sure if this is really a public API.

Reply:
Doesn't the updated `checkValue` of `spark.shuffle.spill.diskWriteBufferSize` already guarantee this?

Reply:
Yes, it can guarantee this.
Here explains why it must be greater than 12.

Reply:
If just for explaining it, maybe you can put a comment on `diskWriteBufferSize` is defined, instead of an `assert`.

Reply:
Where is the best place of this comment? I am neutral on this.